src/types/maybe_undefined.rs

```diff
@@ -0,0 +1,89 @@
+use crate::{registry, InputValueResult, InputValueType, Type, Value};
+use std::borrow::Cow;
+
+/// Similar to `Option`, but it has three states, `undefined`, `null` and `x`.
+///
+/// Spec: https://spec.graphql.org/June2018/#sec-Null-Value
+#[allow(clippy::option_option)]
+pub struct MaybeUndefined<T>(Option<Option<T>>);
```

Comment:
What about making it an enum ?
```rust
enum MaybeUndefined<T> {
    Undefined,
    Null,
    Value(T)
}
```
This would allow for easier pattern matching instead of `if is_null esle if is undefined else ...`